dataInterface/items.js

```diff
@@ -9,15 +9,19 @@ module.exports.getAll = () => {
 }
 
 module.exports.getById = (itemId) => {
-  // TODO: complete writing this function
+  return module.exports.items.find(item => item.id == itemId);
 }
 
 module.exports.deleteById = async (itemId) => {
-    // TODO: complete writing this function
+  const index = module.exports.items.findIndex(item => item.id == itemId);
+  return module.exports.items.splice(index, 1);
 }
 
 module.exports.updateById = async (itemId, newObj) => {
-    // TODO: complete writing this function
+  const index = module.exports.items.findIndex(item => item.id == itemId);
+  if (index > -1) {
+    return module.exports.items[index].field = newObj.field;
+  }
```

Comment:
This method would be improved by having an explicit return value if the item is not found. As is, it has an implicit return of `undefined`, which works with the `router.put` check. However, the code would be easier to understand if after line 24, this method returned null or similar.